service/src/test/java/uk/gov/hmcts/reform/fpl/utils/IncrementalIntegerTest.java

```diff
@@ -0,0 +1,19 @@
+package uk.gov.hmcts.reform.fpl.utils;
+
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class IncrementalIntegerTest {
+    @Test
+    void shouldGetAndIncrementCurrentIntegerValue() {
+        IncrementalInteger i = new IncrementalInteger();
+        assertThat(i.getAndIncrement()).isZero();
+    }
+
+    @Test
+    void shouldIncrementAndGetCurrentIntegerValue() {
+        IncrementalInteger i = new IncrementalInteger();
+        assertThat(i.incrementAndGet()).isEqualTo(1);
```

Comment:
Again this test doesn't check if the increment happened. I can implement this method as value + 1 and the test won't be broken.